pdc/apps/common/tests.py

```diff
@@ -673,3 +676,119 @@ def test_ordering_bad_key(self):
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
         self.assertTrue('Unknown query key' in response.data.get('detail'))
         self.assertTrue('name_' in response.data.get('detail'))
+
+
+class NotificationMixinTestCase(TestCase):
+
+    def setUp(self):
+
+        class DummySerializer(serializers.Serializer):
+            nickname = serializers.CharField()
+            color = serializers.CharField()
+
+            def create(self, validated_data):
+                return mock.Mock(pk=2, **validated_data)
+
+            def update(self, obj, validated_data):
+                return mock.Mock(pk=obj.pk, **validated_data)
+
+        class GetDogMixin(object):
+            def get_object(self):
+                return mock.Mock(pk=1, nickname='Spot', color='black')
+
+        class DummyView(viewsets.NotificationMixin,
+                        mixins.CreateModelMixin,
+                        mixins.UpdateModelMixin,
+                        mixins.DestroyModelMixin,
+                        GetDogMixin,
+                        GenericViewSet):
+            serializer_class = DummySerializer
+
+        mapping = {
+            'post': 'create',
+            'put': 'update',
+            'delete': 'destroy',
+        }
+        self.view = DummyView.as_view(mapping, basename='dummy')
+
+    def _mk_request(self, method, data=None, comment=None):
+        request = django.http.HttpRequest()
+        request.method = method
+        request._messagings = []
+        request.META = {'HTTP_PDC_CHANGE_COMMENT': comment, 'SERVER_NAME': '0.0.0.0', 'SERVER_PORT': 80}
+        if data:
+            data = json.dumps(data)
+            request._read_started = False
+            request.META.update({'CONTENT_LENGTH': len(data), 'CONTENT_TYPE': 'application/json'})
+            request._stream = BytesIO(data)
+        return request
+
+    @mock.patch('pdc.apps.common.viewsets.reverse')
+    def test_create_notification(self, mock_reverse):
+        request = self._mk_request('POST', {'nickname': 'Rover', 'color': 'brown'},
+                                   comment='Sit!')
+        mock_reverse.return_value = '/dogs/2/'
+
+        with mock.patch('pdc.apps.common.viewsets.router') as mock_router:
```

Comment:
Can you also use decorator mock.patch for this?

Reply:
Good suggestion.